ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -860,16 +879,23 @@ def set_waypoints_velocity(self):
                 # Todo evaluate if we need to keep going
                 rospy.logdebug("Now within tl_buffer = {:4.3f}"
                               .format(self.dyn_tl_buffer))
-                if self.waypoints[self.final_waypoints_start_ptr].get_v() <=\
-                        self.handoff_velocity: 
+                #if self.waypoints[self.final_waypoints_start_ptr].get_v() <=\
+                #        self.handoff_velocity: 
+                if self.velocity <= self.min_moving_velocity:
                     self.set_stopped(self.final_waypoints_start_ptr, self.
                                  lookahead_wps)
                 else:
                     rospy.logdebug("Within buffer, but not travelling at creeping speed")
+                    
                     recalc = self.produce_slowdown(self.final_waypoints_start_ptr,
                         self.lookahead_wps,
                         dist_to_tl - (self.dyn_tl_buffer - 1.0))
 
+            elif self.state == 'slowdown' and self.stop_target == self.next_tl_wp:
+                    recalc = self.produce_slowdown(self.final_waypoints_start_ptr,
+                                               self.lookahead_wps,
+                                               dist_to_tl - (self.dyn_tl_buffer - 1.0))
+
```

Comment:
this keeps the car in a fixed slowdown curve and prevents it from popping into speedup if the stopping distance becomes shorter than the distance to the tl_wp